microcosm_postgres/factories/sessionmaker.py

```diff
@@ -1,9 +1,47 @@
+from microcosm.api import defaults
+from microcosm.config.types import boolean
+from microcosm.config.validation import typed
+from microcosm.scoping.factories import ScopedFactory
+from microcosm.scoping.proxies import ScopedProxy
+from sqlalchemy.orm import Session
 from sqlalchemy.orm import sessionmaker
 
 
+ENGINE = "__engine__"
+
+
+def unwrap(bind):
+    return bind.__component__ if isinstance(bind, ScopedProxy) else bind
+
+
+@defaults(
+    # enable multiple engines (binds) via ScopedFactory
+    multi=typed(boolean, default_value=True)
+)
 def configure_sessionmaker(graph):
     """
     Create the SQLAlchemy session class.
 
     """
-    return sessionmaker(bind=graph.postgres)
+    if graph.config.sessionmaker.multi:
+        ScopedFactory.infect(graph, "postgres")
+
+    class RoutingSession(Session):
+        """
+        Route session bind to an appropriate engine.
+
+        See: http://docs.sqlalchemy.org/en/latest/orm/persistence_techniques.html#partitioning-strategies
+
+        """
+        def get_bind(self, mapper=None, clause=None):
```

Comment:
My implementation of `get_bind` implements a strategy based on the model class being used. This is not actually what we want, but it does demonstrate the ability.